src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx

```diff
@@ -0,0 +1,1173 @@
+{
+  std::array<double, 99> fyp
+  {0.0272,0.0326,0.0390,0.0464,0.0551,0.0651,0.0766,0.0898,0.1049,
+   0.1221,0.1416,0.1636,0.1883,0.2159,0.2466,0.2807,0.3182,0.3595,
+   0.4045,0.4535,0.5066,0.5637,0.6249,0.6901,0.7593,0.8324,0.9090,
+   0.9890,1.0720,1.1577,1.2454,1.3349,1.4254,1.5163,1.6070,1.6968,
+   1.7849,1.8705,1.9529,2.0313,2.1049,2.1731,2.2350,2.2901,2.3379,
+   2.3776,2.4090,2.4317,2.4454,2.4500,2.4454,2.4317,2.4090,2.3776,
+   2.3379,2.2901,2.2350,2.1731,2.1049,2.0313,1.9529,1.8705,1.7849,
+   1.6968,1.6070,1.5163,1.4254,1.3349,1.2454,1.1577,1.0720,0.9890,
+   0.9090,0.8324,0.7593,0.6901,0.6249,0.5637,0.5066,0.4535,0.4045,
+   0.3595,0.3182,0.2807,0.2466,0.2159,0.1883,0.1636,0.1416,0.1221,
+   0.1049,0.0898,0.0766,0.0651,0.0551,0.0464,0.0390,0.0326,0.0272};
+   
+  std::array<double, 99> xxp
+  {-3.000,-2.939,-2.878,-2.816,-2.755,-2.694,-2.633,-2.571,-2.510,
+   -2.449,-2.388,-2.327,-2.265,-2.204,-2.143,-2.082,-2.020,-1.959,
+   -1.898,-1.837,-1.776,-1.714,-1.653,-1.592,-1.531,-1.469,-1.408,
+   -1.347,-1.286,-1.224,-1.163,-1.102,-1.041,-0.980,-0.918,-0.857,
+   -0.796,-0.735,-0.673,-0.612,-0.551,-0.490,-0.429,-0.367,-0.306,
+   -0.245,-0.184,-0.122,-0.061, 0.000, 0.061, 0.122, 0.184, 0.245,
+    0.306, 0.367, 0.429, 0.490, 0.551, 0.612, 0.673, 0.735, 0.796,
+    0.857, 0.918, 0.980, 1.041, 1.102, 1.163, 1.224, 1.286, 1.347,
+    1.408, 1.469, 1.531, 1.592, 1.653, 1.714, 1.776, 1.837, 1.898,
+    1.959, 2.020, 2.082, 2.143, 2.204, 2.265, 2.327, 2.388, 2.449,
+    2.510, 2.571, 2.633, 2.694, 2.755, 2.816, 2.878, 2.939, 3.000};
+   
+  double MN = fPM;
+  double MN2 = MN*MN;
+  double Mp = fMP;
+  double Mp2 = Mp*Mp;
+  double W2 = W*W;
+  
+  double nu = (W2 - MN2 + Q2)/2./MN;
+  double qv = TMath::Sqrt(nu*nu + Q2);
+  // assume this is 2*pf*qv
+  double dW2dpF = 2.*qv;
+  double dW2dEs = 2.*(nu + MN); 
+  // switched to using 99 bins!
+
+  for (int i=0; i<100; i++)
+  {
+    double fyuse = fyp[i+1]/100.;
+    double W2p = W2 + xxp[i+1]*pF*dW2dpF - Es*dW2dEs;
+    if(W2p>1.159)
+    {
+       //proton 
+       double Wp = TMath::Sqrt(W2p);
+       double sigmaTp = sigmaR(0, Q2, Wp) + sigmaNR(0, Q2, Wp);
+       double sigmaLp = sigmaR(1, Q2, Wp) + sigmaNR(1, Q2, Wp); 
+       double F1pp = sigmaTp*(W2p-Mp2)/8./kPi2/kAem;
+       //neutron
+       double sigmaTd = sigmaR(0, Q2, Wp, true) + sigmaNR(0, Q2, Wp, true);
+       double F1dp = sigmaTd*(W2p-Mp2)/8./kPi2/kAem;
+       F1d += F1dp*fyuse;
+       F1p += F1pp*fyuse;
+       sigmaT += sigmaTp*fyuse;
+       sigmaL += sigmaLp*fyuse;
+    }
+  }
+
+}
+//___________________________________________________________________________
+// Calculate proton and neutron with Fermi smearing of a deuteron 
+void BostedChristyEMPXSec::FermiSmearingD(double Q2, double W, double & F1, double & R, double & sigmaT, double & sigmaL, bool isDeuterium=false) const
+{
+  std::array<double, 20> fyd
```

Comment:
After all the discussion we had about configurable parameters, I'm sure these hard-coded arrays have a reason to be so. Can you add a comment in the file about what they are and why they are not supposed to change?
It a few years from now it will help us to read this code.

Reply:
From the technical point of view, maybe it's a good idea to define them as static and constexpr.